src/python/grpcio/grpc/beta/_server_adaptations.py

```diff
@@ -161,14 +161,24 @@ def draw_all_values(self):
           self._condition.wait()
 
 
-def _pipe_requests(request_iterator, request_consumer, servicer_context):
-  for request in request_iterator:
-    if not servicer_context.is_active():
-      return
-    request_consumer.consume(request)
-    if not servicer_context.is_active():
-      return
-  request_consumer.terminate()
+def _run_request_pipe_thread(request_iterator, request_consumer,
+                             servicer_context):
+  thread_joined = (False,)
+  def pipe_requests():
+    for request in request_iterator:
+      if not servicer_context.is_active() or thread_joined[0]:
+        return
+      request_consumer.consume(request)
+      if not servicer_context.is_active() or thread_joined[0]:
+        return
+    request_consumer.terminate()
+
+  def stop_request_pipe(timeout):
+    thread_joined[0] = True
```

Comment:
My guess would be this gets exercised by the interop cancel_after_first_response test, and that is why it is failing.  I don't have any unit tests for the beta scenario however.